src/utils/references.ts

```diff
@@ -44,3 +45,47 @@ export function stringifyReferences (paths: string[], cwd: string): string {
 export function toReference (path: string, cwd: string): string {
   return `/// <reference path="${normalizeSlashes(isAbsolute(path) ? relative(cwd, path) : normalize(path))}" />`
 }
+
+export enum DependencyTarget {
+  Main,
+  Browser
+}
+
+export enum DependencyType {
+  Ambient,
+  External
+}
+
+export interface DependencyInfo {
+  target: DependencyTarget
+  type: DependencyType
+  name: string
+}
+
+export function parseReferencePath(path: string): DependencyInfo {
+  const referencePathRegexPattern =
```

Comment:
I'm thinking about moving this parse function into `prune.ts` since it doesn't seem like anybody else is going to need to reverse parse a reference path. Thoughts?